src/Umbraco.Web/PropertyEditors/LabelPropertyEditor.cs

```diff
@@ -43,32 +41,94 @@ public override bool IsReadOnly
             {
                 Fields.Add(new PreValueField()
                 {
                     HideLabel = true,
                     View = "readonlykeyvalues",
-                    Key = "values"
+                    Key = LegacyPropertyEditorValuesKey
                 });
+
+                ValueType = PropertyEditorValueTypes.StringType;
             }
 
+            [PreValueField(Constants.PropertyEditors.DataValueTypePreValueKey, "Value type", "valuetype")]
+            public string ValueType { get; set; }
+
             /// <summary>
-            /// Chuck all the values into one field so devs can see what is stored there - we want this in case we've converted a legacy proeprty editor over to a label
-            /// we should still show the pre-values stored for the data type.
+            /// Other than for the pre-value fields defined on this property editor, chuck all the values into one field so devs can see what is stored there.
+            /// We want this in case we've converted a legacy property editor over to a label as we should still show the pre-values stored for the data type.
             /// </summary>
             /// <param name="defaultPreVals"></param>
             /// <param name="persistedPreVals"></param>
             /// <returns></returns>
             public override IDictionary<string, object> ConvertDbToEditor(IDictionary<string, object> defaultPreVals, PreValueCollection persistedPreVals)
             {
                 var existing = base.ConvertDbToEditor(defaultPreVals, persistedPreVals);
-                //convert to a list, easier to enumerate on the editor
-                var asList = existing.Select(e => new KeyValuePair<string, object>(e.Key, e.Value)).ToList();
-                var result = new Dictionary<string, object> { { "values", asList } };
+
+                // Check for a saved value type.  If not found set to default string type.
+                var valueType = PropertyEditorValueTypes.StringType;
+                if (existing.ContainsKey(Constants.PropertyEditors.DataValueTypePreValueKey))
+                {
+                    valueType = (string)existing[Constants.PropertyEditors.DataValueTypePreValueKey];
+                }
+
+                // Convert any other values from a legacy property editor to a list, easier to enumerate on the editor.
+                // Make sure to exclude values defined on the label property editor itself.
+                var asList = existing
+                    .Select(e => new KeyValuePair<string, object>(e.Key, e.Value))
+                    .Where(e => e.Key != Constants.PropertyEditors.DataValueTypePreValueKey)
+                    .ToList();
+
+                var result = new Dictionary<string, object> { { Constants.PropertyEditors.DataValueTypePreValueKey, valueType } };
+                if (asList.Any())
+                {
+                    result.Add("values", asList);
+                }
+

```

Comment:
Previous code was always sending "values" even when the list was empty. Is it OK to not send it if empty? Also is "values" here `LegacyPropertyEditorValuesKey` or is it just a coincidence?